cdc/processor/pipeline/sink.go

```diff
@@ -303,21 +300,27 @@ func (n *sinkNode) clearBuffers() {
 	}
 }
 
-func (n *sinkNode) emitRow2Sink(ctx context.Context) error {
+// emitRow2Sink emit rows in sinkNode.rowBuffer to backend sink (table sink in this case).
+// If it returns false and error is nil, the caller should retry until it returns true.
+// Otherwise, the rows will be lost.
+func (n *sinkNode) emitRow2Sink(ctx context.Context) (bool, error) {
 	for _, ev := range n.eventBuffer {
 		n.rowBuffer = append(n.rowBuffer, ev.Row)
 	}
+	n.eventBuffer = n.eventBuffer[:0]
```

Comment:
This can delay memory being released. Can you recreate the event buffer?

Reply:
Actually sinkNode.eventBuffer is redundant, I will remove it in next PR.